third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -2207,8 +2207,7 @@ impl Vector {
         Self::pack(type_param, vec![])
     }
 
-    pub fn unpack(self, type_param: &Type, expected_num: u64) -> PartialVMResult<Vec<Value>> {
-        check_elem_layout(type_param, &self.0)?;
+    pub fn unpack_unchecked(self) -> PartialVMResult<Vec<Value>> {
```

Comment:
Why is it okay to do this unchecked operation?

Reply:
I think the semantics of unpack is kind of ridiculous. We have a function to unpack a move-vector into a vector of move values, that needs us to give the number of elements just for a check. There is no actual way to get the number of elements, so it's all useless

Reply:
@alinush I think this is fine.

The reason why we only had the checked `unpack` function was because at the time we implemented this, we didn't not want to open up values to clients outside the core VM (native functions included), and the `unpack` function we had was directly mimicking the `vec_unpack` instruction.

Nowadays there's a practical need to use values inside native functions, and converting a vector value into the Rust version is one of such reasonable use cases.